### ISSUE TITLE
passenger-status: the --verbose option does not work

### ISSUE THREAD

**Issue by majgis** [phusion/passenger#1716]
The passenger-status --verbose option is not working:

```
$ sudo passenger-status --verbose
Version : 5.0.22
Date    : Wed Dec 30 18:54:31 -0700 2015
Instance: o483SdDy (nginx/1.8.0 Phusion_Passenger/5.0.22)

----------- General information -----------
Max pool size : 1
App groups    : 1
Processes     : 1
Requests in top-level queue : 0

----------- Application groups -----------
/opt/testApp/node_modules/testApp/public (production):
  App root: /opt/testApp/node_modules/testApp
  Requests in queue: 0
  * PID: 23090   Sessions: 0       Processed: 12      Uptime: 8h 27m 44s
    CPU: 0%      Memory  : 36M     Last used: 5h 41m 3

```

Here is the documentation showing more information when using verbose:
https://www.phusionpassenger.com/library/admin/standalone/request_individual_processes.html

This is for a Node.js app on stand-alone passenger.

**Comment by OnixGH** [comment]
At the top of the documentation there is a yellow note stating "This feature is currently available for Ruby apps only."

**Comment by majgis** [comment]
Can we open another issue then, asking to make this existing feature work for node.js apps?  

I use `passenger-status --verbose --show=xml` and I get a socket address for each pid, but no password.  When I hit that socket address, I get permission denied, so I assume I need so set the password in the header like in the documentation example...but nowhere is the password available.

Looking at the source code for passenger-source, I notice it hits some core api.  Can you make a suggestion on how I can extract pid,url(http or socket),password?

I can think of these approaches:
1. Add password to socket section of `passenger-status --verbose --show=xml`
2. Make `passenger-status --verbose` work for node.js apps
3. Write a hackalicious ruby script based on what I see in passenger-status

Please advise on how I might extract this information, thank you.

**Comment by FooBarWidget** [comment]
Passenger makes the Node.js app listen on a Unix domain socket file. This file is protected through Unix filesystem permissions. Unlike in Ruby, no connect password is used.

So it is possible to connect directly to a specific Node.js process, but you will need an HTTP client that supports Unix domain sockets. I think newer versions of curl supports that. Otherwise, you can also write your own HTTP client using raw sockets.

If you get a permission denied error, then that's a filesystem permission error, not an error related to the connect password. Try again with root privileges.

**Comment by majgis** [comment]
I was so close in my first test, but I foolishly used sudo on the echo command, thinking it applied to socat.  The following command worked:

    $ passenger-status --show=xml -v #shows <address>unix:/tmp/passenger.30Ge2aS/apps.s/node.34l02e</address>
    $ echo -e "GET /health HTTP/1.1\r\n" | sudo socat unix-connect:/tmp/passenger.30Ge2aS/apps.s/node.34l02e STDIO
    HTTP/1.1 200 OK
    X-Powered-By: Express
    Content-Type: text/html; charset=utf-8
    Content-Length: 16
    ETag: W/"10-8MtBq0wHBRF66BmScI50YQ"
    Date: Mon, 04 Jan 2016 17:42:33 GMT
    Connection: keep-alive

    All systems go!

I used socat because curl was not available on centos 6.5, which I was testing against.

Thank you, this information was a great help, and solved my issue.
